webapp/src/app/admin/devices/devices.component.ts

```diff
@@ -53,22 +55,31 @@ export class DevicesComponent implements OnInit {
             return showDialogInfo(this.dialog, "No Suicide", "Cannot delete one's own device for security " +
                 "reasons.", "Understood");
         }
-        await showTransactions(this.dialog, "Deleting device " + device.name,
-            async (progress: TProgress) => {
-                progress(30, "Deleting Device");
-                await this.uData.contact.deleteDevice(device.name);
-                progress(60, "Updating Device List");
-                await this.uData.contact.sendUpdate();
-                progress(-90, "Fetching all devices");
-                this.updateDevices();
-            });
+        if (await showDialogOKC(this.dialog, `Deleting device`,
+            `Do you really want to delete the device ${device.name}?`,
+            {OKButton: `Delete`, CancelButton: `Keep`})) {
+            await showTransactions(this.dialog, "Deleting device " + device.name,
+                async (progress: TProgress) => {
+                    progress(30, "Deleting Device");
+                    await this.uData.contact.deleteDevice(device.name);
+                    progress(60, "Updating Device List");
+                    await this.uData.contact.sendUpdate();
+                    progress(-90, "Fetching all devices");
+                    this.updateDevices();
+                });
+        }
+
     }
 
-    async add() {
+    async addDevice() {
         const ac = this.dialog.open(DeviceAddComponent);
         ac.afterClosed().subscribe(async (result) => {
             if (result) {
```

Comment:
```suggestion
             if (result === undefined || result === "") {
```

Reply:
Please take care when proposing these changes. I saw several inverted suggestions from your part, and one took me quite some time to discover afterwards...